src/components/LeadsTable/leadsSlice.ts

```diff
@@ -0,0 +1,119 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+interface Lead {
+   _id: string;
+   name: string;
+   email: string;
+   consentsAccepted: boolean;
+   createdAt: string;
+   updatedAt: string;
+   __v: number;
+}
+
+interface LeadsState {
+   leads: Lead[];
+   isLoading: boolean;
+   error: string | null;
+   sortKey: 'name' | 'email' | 'date' | 'consentsAccepted';
+   sortOrder: 'asc' | 'desc';
+}
+
+const initialState: LeadsState = {
+   leads: [],
+   isLoading: false,
+   error: null,
+   sortKey: 'date',
+   sortOrder: 'desc',
+};
+
+export const fetchLeads = createAsyncThunk('leads/fetchLeads', async () => {
+   const response = await fetch('https://training.nerdbord.io/api/v1/leads');
+   const data = await response.json();
+   return data;
+});
+
+export const sortLeadsByColumn = createAsyncThunk(
+   'leads/sortLeadsByColumn',
```

Comment:
**issue:** Sorting does not need to be a part of `leadsSlice` - it's more UI responsible part than data layer.